server/repository/src/db_diesel/patient.rs

```diff
@@ -197,6 +200,11 @@ impl<'a> PatientRepository<'a> {
                 query = query.or_filter(name_dsl::id.eq_any(sub_query))
             }
 
+            if name_and_code.is_some() {
+                apply_string_filter!(query, name_and_code.clone(), name_dsl::name_);
+                apply_string_or_filter!(query, name_and_code.clone(), name_dsl::code);
+            }
```

Comment:
is this working in combination with the identifier? there are some limitations with how the ORs work and I have the suspicion that it doesn't work when using the identifier and the new filer... Could you please add a test?

You could also check the generated SQL.

Reply:
[fa9af9a](https://github.com/openmsupply/open-msupply/pull/2089/commits/fa9af9a52a6d306b58dd4cd5b74eb211d8170d38)

Reply:
I created an issue for a case where it goes wrong...
https://github.com/openmsupply/open-msupply/issues/2109

Maybe we can extend the identifier filter to include the name?